devito/mpi/halo_scheme.py

```diff
@@ -28,7 +28,39 @@ class HaloLabel(Tag):
 STENCIL = HaloLabel('stencil')
 
 
-HaloSchemeEntry = namedtuple('HaloSchemeEntry', 'loc_indices loc_dirs halos dims')
+class HaloSchemeEntry:
+
+    def __init__(self, loc_indices, loc_dirs, halos, dims):
+        self.loc_indices = loc_indices
+        self.loc_dirs = loc_dirs
+        self.halos = halos
+        self.dims = dims
+
+    def __repr__(self):
+        return (f"HaloSchemeEntry(loc_indices={self.loc_indices}, "
+                f"loc_dirs={self.loc_dirs}, halos={self.halos}, dims={self.dims})")
+
+    def __eq__(self, other):
+        if not isinstance(other, HaloSchemeEntry):
+            return False
+        return (self.loc_indices == other.loc_indices and
+                self.loc_dirs == other.loc_dirs and
+                self.halos == other.halos and
+                self.dims == other.dims)
+
+    def __hash__(self):
```

Comment:
This was autocompleted by copilot.
Are people happy to have class instead of namedtuple?